features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -187,7 +187,24 @@ internal class SessionReplayFeature(
             return
         }
 
-        if (keepSession && rateBasedSampler.sample()) {
+        val recordNewSession = keepSession && rateBasedSampler.sample()
+
+        if (!initialized.get()) {
+            sdkCore.internalLogger.log(
+                InternalLogger.Level.WARN,
+                InternalLogger.Target.USER,
+                {
+                    if (recordNewSession) {
+                        CANNOT_START_RECORDING_NOT_INITIALIZED
+                    } else {
+                        CANNOT_STOP_RECORDING_NOT_INITIALIZED
+                    }
+                }
+            )
```

Comment:
Here I can use a single message (not making a difference between start and stop) or move this to a dedicated method if this is too verbose for this function.

Reply:
If we use a single message I can even move this check above in the function